src/api/app/controllers/status_project_controller.rb

```diff
@@ -0,0 +1,42 @@
+class StatusProjectController < ApplicationController
+  def show
+    dbproj = Project.get_by_name(params[:project])
+    @packages = ProjectStatus::Calculator.new(dbproj).calc_status
+    find_relationships_for_packages(@packages)
+  end
+
+  private
+
+  def role_from_cache(role_id)
+    @rolecache[role_id] || (@rolecache[role_id] = Role.find(role_id).title)
```

Comment:
This could be `@rolecache[role_id] ||= Role.find(role_id).title`

Same for the other `*_from_cache` methods